### ISSUE TITLE
Segmentation Fault

### ISSUE THREAD

**Issue by ethanmancini** [ethanmancini/SPICE-2#1]
A segmentation fault occurs after a simulation is loaded.

**Comment by ethanmancini** [comment]
The error occurs after the temperature is displayed. 
`Backtrace for this error:
#0  0x7F40E92EE467
#1  0x7F40E92EEAAE
#2  0x7F40E87F566F
#3  0x451AB9 in comprs_
#4  0x420CA9 in extmem_
#5  0x459B09 in topchk_
#6  0x45369F in errchk_
#7  0x468197 in MAIN__ at spice.f:?
`

**Comment by tvrusso** [comment]
Not sure if this is the same issue as I found, but I'm pretty sure that the "loc_" function in unix.c is assuming that addresses and ints are the same size, and that is really unlikely to be true on any modern system.  I found that the code segfaults in comprs because the result of doing a cast from an int* to an int is getting negative numbers, and these are then used to compute offsets into a fortran common block.

I tried to play around with forcing gfortran to build with 8 bit integers instead (adding the "-finteger-4-integer-8" to the compiler options), and changing the loc_ function in unix.c to use long ints instead, and this stopped the segfaults, but there is still something very wrong and the code now exits cleanly with errors about table pointers being invalid.   There are at least several other places where integers are assumed to be four bytes long, including the NXTEVN function.

I was unable in the few minutes I had to spare on this to track down the others.  I tried several things to try to change the assumption from 4 to 8 everywhere I found it (LOCF and NXTEVN), but while this stopped the segfaults, it did not actually make the program work.

It sure would be fun to band-aid SPICE2 into working on a modern computer, even though there are plenty of modern alternatives to use instead.
